contents/handbook/engineering/query-performance-optimization.md

```diff
@@ -1,28 +1,82 @@
 ---
-title: ClickHouse query performance optimization
+title: Query performance optimization
 sidebar: Handbook
 showTitle: true
 ---
 
 Making sure PostHog operates fast at scale is key to our success.
 
-This document outlines some tools we have to discover and fix issues.
+This document outlines some best practices to archive good query performance at scale, as well as describing tools and procedures to discover and fix performance issues.
 
-# Tools of the trade
+PostHog uses two different datastores:
 
-## 1. Grafana "Clickhouse queries - by endpoint" dashboard
+- **PostgreSQL**: row-oriented OLTP database, mainly used to store and query datasets using predictable clausole(s). It is likely your best choice if:
+    - the query pattern to access your dataset is predictable
+    - the dataset will likely not grow overtime above (<= 1 TB)
+    - the dataset needs to mutate often (`DELETE`/`UPDATE`)
+    - the query pattern requires joins across multiple tables
 
-https://metrics.posthog.com/d/vo7oCVZ7z/clickhouse-queries-by-endpoint
+- **ClickHouse**: column-oriented OLAP database, used to store large datasets and run on them analytical queries. It is likely your best choice if:
+    - the query pattern to access your dataset is unpredictable
+    - the dataset will likely grow overtime (> 1 TB)
+    - the dataset doesn't need to mutate often (`DELETE`/`UPDATE`)
+    - the query pattern doesn't requires joins across multiple tables
 
-This dashboard gives a breakdown of how things are looking reliability and performance-wise.
+## PostgreSQL
 
+#### Coding best practices
```

Comment:
Is it possible that we also include some equivalent Django ORM examples. Or just a comment to say also consider these points with the ORM? If in doubt you can have a look at the generated sql by doing X

Reply:
Done